### PR TITLE
LRQA-66115 | master

### PR DESCRIPTION
**Pull request by melodywu**
https://issues.liferay.com/browse/LRQA-66115

### CODE REVIEW

**Comment by melodywu** [comment]
ci:test:sf

**Comment by melodywu** [comment]
ci:test:sf

**Comment by Sharryxue** [comment]
Just starting review.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase [hunk 1]
```diff
@@ -735,6 +735,65 @@ definition {
 			value1 = "Error:Please upload a file with a valid extension (JAR, LPKG, or WAR File).");
 	}
 
+	@description = "This test check search apps in marketplace store and check sign in and sign out."
+	@priority = "3"
+	test SearchMarketplaceStore {
+		task ("When user login marketplace store assert user login in Purchased page") {
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase:741]: Please start with Given in task when applying the BDD mode. Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase [hunk 2]
```diff
@@ -841,4 +841,59 @@ definition {
 		AssertElementNotPresent(locator1 = "Portlet#HEADER_DUPLICATE");
 	}
 
+	@description = "This test view purchased apps in marketplace purchased page."
+	@priority = "3"
+	test ViewPurchasedApp {
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase:846]: Seems this 2 tests can be combined. The previous can be partial of this one. Could you try to combining the steps? Thanks in advance.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase [hunk 3]
```diff
@@ -841,4 +841,59 @@ definition {
...
+				panel = "Control Panel",
+				portlet = "Store");
+
+			WaitForPageLoad();
+
+			AppManager.searchApp(searchName = "Control Panel Documentation");
+
+			WaitForPageLoad();
+
+			AppManager.purchaseApp(
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase:879]: See if this is duplicate with the filter by price test. Thanks

**Review by Sharryxue — CHANGES_REQUESTED** [review]
Hello @melodywu,
Could you please double check if these 2 tests can be combined, and see if the Filter by Price tests can also be covered here? Thanks in advance.
BTW, the BDD model start with Given.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase (hunk 1 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase:741]: combine search and filter by price tests

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase (hunk 2 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/controlpanel/apps/appmanager/AppManagement.testcase:846]: ok and done

**Comment by Sharryxue** [comment]
Resend [Sharryxue/liferay-portal#421](https://github.com/Sharryxue/liferay-portal/pull/420).

**Pull request closed without merging**